src/clients/scan-api/operations/v0/registry/allocation/v2/get-settlement-factory-from-registry.ts

```
@@ -0,0 +1,85 @@
+    readonly choiceContextData: GetSettlementFactoryFromRegistryChoiceContextData;
+    readonly disclosedContracts: readonly GetSettlementFactoryFromRegistryDisclosedContract[];
+  };
+}
+
+function buildRegistryEndpoint(registryUrl: string): string {
+  const url = new URL(registryUrl);
+  url.username = '';
+  url.password = '';
+  url.search = '';
+  url.hash = '';
+  url.pathname = `${url.pathname.replace(/\/+$/, '')}${endpoint}`;
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)